offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/offers/OverviewService.groovy

```diff
@@ -15,40 +16,76 @@ import life.qbic.portal.offermanager.dataresources.offers.OfferOverview
  *
  * @since 1.0.0
  */
-class OverviewService implements ResourcesService {
+class OverviewService implements ResourcesService<OfferOverview> {
 
     private List<OfferOverview> offerOverviewList
 
     private final OfferDbConnector offerDbConnector
 
     private final OfferResourcesService offerService
 
-    final EventEmitter<String> updatedOverviewEvent
+    private final EventEmitter<OfferOverview> updatedOverviewEvent
 
     OverviewService(OfferDbConnector offerDbConnector,
                     OfferResourcesService offerService) {
         this.offerDbConnector = offerDbConnector
         this.updatedOverviewEvent = new EventEmitter<>()
         this.offerService = offerService
-        this.offerOverviewList = []
+        this.offerOverviewList = offerDbConnector.loadOfferOverview()
         subscribeToNewOffers()
-        reloadResources()
     }
 
     private void subscribeToNewOffers(){
         /*
         Whenever a new offer is created, we want
         to update the offer overview content.
          */
-        offerService.offerCreatedEvent.register({
-            reloadResources()
-            updatedOverviewEvent.emit("New overview content available!")
+        offerService.subscribe({
+            def newOfferOverview = createOverviewFromOffer(it)
+            offerOverviewList.add(newOfferOverview)
+            updatedOverviewEvent.emit(newOfferOverview)
         })
     }
 
+    static OfferOverview createOverviewFromOffer(it) {
```

Comment:
```suggestion
    static OfferOverview createOverviewFromOffer(Offer it) {
```
since this is a public method the parameter type should be explicit :)